File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/encryption_decryption/Contract.java
package gateway.hitrontech.com.encryption.fragment.encryption_decryption;

import moe.xing.mvp_utils.BasePresenter;
import moe.xing.mvp_utils.BaseView;

public interface Contract {

  interface Presenter extends BasePresenter {

    void getCipherText(String key, String plainText);

    void getPlainText(String key, String cipherText);

  }

  interface View extends BaseView<Presenter> {

    void setCipherText(String cipherText);

    void setPlainText(String plainText);
  }

}

File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/encryption_decryption/Contract.kt
package gateway.hitrontech.com.encryption.fragment.encryption_decryption

import moe.xing.mvp_utils.BasePresenter
import moe.xing.mvp_utils.BaseView

interface Contract {

    interface Presenter : BasePresenter {

        fun getCipherText(key: String, plainText: String)

        fun getPlainText(key: String, cipherText: String)

    }

    interface View : BaseView<Presenter> {

        fun setCipherText(cipherText: String)

        fun setPlainText(plainText: String)
    }

}
